fog/ledger/server/tests/router_connection.rs

```diff
@@ -0,0 +1,908 @@
+            client_auth_token_max_lifetime: Default::default(),
+            query_retries: 3,
+            omap_capacity: OMAP_CAPACITY,
+        };
+
+        let enclave = LedgerSgxEnclave::new(
+            get_enclave_path(mc_fog_ledger_enclave::ENCLAVE_FILE),
+            &config.client_responder_id,
+            OMAP_CAPACITY,
+            logger.clone(),
+        );
+
+        let grpc_env = Arc::new(grpcio::EnvBuilder::new().build());
+
+        let ra_client =
+            AttestClient::new(&config.ias_api_key).expect("Could not create IAS client");
+        let mut ledger_server = LedgerRouterServer::new(
+            config,
+            enclave,
+            ra_client,
+            Arc::new(RwLock::new(HashMap::new())),
+            ledger.clone(),
+            watcher.clone(),
+            logger.clone(),
+        );
+
+        ledger_server.start();
+
+        // Make unattested ledger client
+        let client = FogUntrustedLedgerGrpcClient::new(
+            client_listen_uri,
+            GRPC_RETRY_CONFIG,
+            grpc_env,
+            logger.clone(),
+        );
+
+        // Get a tx_out that is actually in the ledger
+        let real_tx_out0 = { ledger.get_tx_out_by_index(0).unwrap() };
+
+        // Try to get tx out records
+        let queries: Vec<CompressedRistrettoPublic> =
+            vec![(&[0u8; 32]).into(), real_tx_out0.public_key];
+        let result = client.get_tx_outs(queries).unwrap();
+        // Check that we got expected num_blocks value
+        assert_eq!(result.num_blocks, 4);
+        // Check that we got 2 results, as expected
+        assert_eq!(result.results.len(), 2);
+        assert_eq!(
+            &result.results[0].tx_out_pubkey.clone().unwrap().data[..],
+            &[0u8; 32]
+        );
+        assert_eq!(result.results[0].result_code, TxOutResultCode::NotFound);
+        assert_eq!(
+            &result.results[1].tx_out_pubkey.clone().unwrap().data[..],
+            &real_tx_out0.public_key.as_bytes()[..]
+        );
+        assert_eq!(result.results[1].result_code, TxOutResultCode::Found);
+        assert_eq!(result.results[1].tx_out_global_index, 0);
+        assert_eq!(result.results[1].block_index, 0);
+        assert_eq!(
+            result.results[1].timestamp_result_code,
+            TimestampResultCode::BlockIndexOutOfBounds as u32
+        );
+    }
+
+    // grpcio detaches all its threads and does not join them :(
+    // we opened a PR here: https://github.com/tikv/grpc-rs/pull/455
+    // in the meantime we can just sleep after grpcio env and all related
+    // objects have been destroyed, and hope that those 6 threads see the
+    // shutdown requests within 1 second.
+    sleep(Duration::from_millis(1000));
+}
+
+// Infra
+
+/// Adds a block containing one txo for each provided recipient and returns new
+/// block height.
+///
+/// # Arguments
+/// * `ledger_db`
+/// * `recipients` - Recipients of outputs.
+/// * `rng`
```

Comment:
:wrench: Should either fill out all the arguments or remove, having 3 of the 6 arguments listed here with only one documented seems odd.